client/src/endpoint/trades.rs

```diff
@@ -0,0 +1,257 @@
+//! Contains the endpoint for all trades.
+use error::Result;
+use std::str::FromStr;
+use stellar_resources::{Trade, AssetIdentifier};
+use super::{Body, IntoRequest, Order, Records};
+use http::{Request, Uri};
+
+/// Represents the all trades endpoint for the stellar horizon server. The endpoint
+/// will return all trades filtered by a myriad of different query params.
+///
+/// <https://www.stellar.org/developers/horizon/reference/endpoints/trades.html>
+///
+/// ## Example
+/// ```
+/// use stellar_client::sync::Client;
+/// use stellar_client::endpoint::trades;
+///
+/// let client      = Client::horizon_test().unwrap();
+/// let endpoint    = trades::All::default();
+/// let records     = client.request(endpoint).unwrap();
+/// #
+/// # assert!(records.records().len() > 0);
+/// ```
+#[derive(Debug, Default)]
+pub struct All {
+    base_asset: Option<AssetIdentifier>,
+    counter_asset: Option<AssetIdentifier>,
+    offer_id: Option<u32>,
+    cursor: Option<String>,
+    order: Option<Order>,
+    limit: Option<u32>,
+}
+
+impl All {
+    /// /// Fetches the records of a trade for a specified base asset.
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// # extern crate stellar_resources;
+    /// # extern crate stellar_client;
+    /// 
+    /// use stellar_client::sync::Client;
+    /// use stellar_client::endpoint::trades;
+    /// use stellar_resources::AssetIdentifier;
+    /// 
+    /// # fn main() {
+    ///     let base_asset = AssetIdentifier::alphanum4("MOBI", "GA6HCMBLTZS5VYYBCATRBRZ3BZJMAFUDKYYF6AH6MVCMGWMRDNSWJPIH");
```

Comment:
Is this asset guaranteed to exist even if the testnet is refreshed? We've actually started leaving these integration tests out of the function-specific doctests recently actually.